contrib/tracing/log_utxo_coins.py

```diff
@@ -0,0 +1,116 @@
+  u64 usage;
+};
+
+BPF_PERF_OUTPUT(coins_added);
+BPF_PERF_OUTPUT(coins_spent);
+BPF_PERF_OUTPUT(coins_uncached);
+
+int trace_add_coin(struct pt_regs *ctx) {
+  struct coin_data data = {};
+  bpf_usdt_readarg_p(1, ctx, &data.hash, MAX_HASH_SIZE);
+  bpf_usdt_readarg(2, ctx, &data.index);
+  bpf_usdt_readarg(3, ctx, &data.height);
+  bpf_usdt_readarg(4, ctx, &data.value);
+  bpf_usdt_readarg(5, ctx, &data.isCoinBase);
+  bpf_usdt_readarg(6, ctx, &data.usage);
+  coins_added.perf_submit(ctx, &data, sizeof(data));
+  return 0;
+}
+
+int trace_spend_coin(struct pt_regs *ctx) {
+  struct coin_data data = {};
+  bpf_usdt_readarg_p(1, ctx, &data.hash, MAX_HASH_SIZE);
+  bpf_usdt_readarg(2, ctx, &data.index);
+  bpf_usdt_readarg(3, ctx, &data.height);
+  bpf_usdt_readarg(4, ctx, &data.value);
+  bpf_usdt_readarg(5, ctx, &data.isCoinBase);
+  bpf_usdt_readarg(6, ctx, &data.usage);
+  coins_spent.perf_submit(ctx, &data, sizeof(data));
+  return 0;
+}
+
+int trace_uncached_coin(struct pt_regs *ctx) {
+    struct coin_data data = {};
+    bpf_usdt_readarg_p(1, ctx, &data.hash, MAX_HASH_SIZE);
+    bpf_usdt_readarg(2, ctx, &data.index);
+    bpf_usdt_readarg(3, ctx, &data.height);
+    bpf_usdt_readarg(4, ctx, &data.value);
+    bpf_usdt_readarg(5, ctx, &data.isCoinBase);
+    bpf_usdt_readarg(6, ctx, &data.usage);
+    coins_uncached.perf_submit(ctx, &data, sizeof(data));
+    return 0;
+}
+"""
+
+
+def print_event(event, action):
+    # print(event.hash.decode('ASCII'), action, event.height, event.value, "YES" if event.isCoinBase else "NO", event.usage)
+    print("%-70s %-10s %-10d %-15d %-10s %-10d" %
+          (f"{event.hash.decode('ASCII')}:{str(event.index)}", action, event.height, event.value, "YES" if event.isCoinBase else "NO", event.usage))
+
+
+def main(bitcoind_path):
+    bitcoind_with_usdts = USDT(path=str(bitcoind_path))
+    bitcoind_with_usdts.enable_probe(
+        probe="add", fn_name="trace_add_coin")
+    bitcoind_with_usdts.enable_probe(
+        probe="spent", fn_name="trace_spend_coin")
+    bitcoind_with_usdts.enable_probe(
+        probe="uncache", fn_name="trace_uncached_coin")
+
+    b = BPF(text=program, usdt_contexts=[bitcoind_with_usdts])
+
+    def handle_coins_added(_, data, size):
+        event = b["coins_added"].event(data)
+        print_event(event, "Add")
+
+    def handle_coins_spent(_, data, size):
+        event = b["coins_spent"].event(data)
+        print_event(event, "Spent")
+
+    def handle_coins_uncached(_, data, size):
+        event = b["coins_uncached"].event(data)
+        print_event(event, "Uncache")
+
+    b["coins_added"].open_perf_buffer(handle_coins_added)
+    b["coins_spent"].open_perf_buffer(handle_coins_spent)
+    b["coins_uncached"].open_perf_buffer(handle_coins_uncached)
+
+    print("Logging Add, Spend and Uncache in the UTXO set. Ctrl-C to end...")
+    print("%-70s %-10s %-10s %-15s %-10s %s" %
+          ("txHash", "Action", "Height", "Satoshis", "Coin Base", "Usage"))
```

Comment:
```suggestion
          ("Outpoint", "Action", "Height", "Value", "Coinbase", "Memory Usage"))
```

Reply:
Maybe also add cache size (utxo set count)?

Reply:
In src/coins.cpp you should be able to access cacheCoins in src/coins.h https://github.com/bitcoin/bitcoin/blob/7317e14a44c6efc545e6fb9bcedee7174e93a8fa/src/coins.h#L221

which is a `std::unordered_map` https://github.com/bitcoin/bitcoin/blob/7317e14a44c6efc545e6fb9bcedee7174e93a8fa/src/coins.h#L134

which implements size() https://en.cppreference.com/w/cpp/container/unordered_map/size